File: hermes/src/main/java/com/lodz/android/hermes/contract/OnConnectListener.java
package com.lodz.android.hermes.contract;

/**
 * 连接监听器
 * Created by zhouL on 2018/5/23.
 */
public interface OnConnectListener {
    /**
     * 连接完成
     * @param isReconnected 是否属于重连
     */
    void onConnectComplete(boolean isReconnected);

    /**
     * 连接失败
     * @param cause 异常
     */
    void onConnectFailure(Throwable cause);

    /**
     * 连接丢失（连上后断开）
     * @param cause 异常
     */
    void onConnectionLost(Throwable cause);
}

File: hermes/src/main/java/com/lodz/android/hermes/contract/OnConnectListener.kt
package com.lodz.android.hermes.contract

/**
 * 连接监听器
 * @author zhouL
 * @date 2019/12/20
 */
interface OnConnectListener {

    /** 连接完成，是否属于重连[isReconnected] */
    fun onConnectComplete(isReconnected: Boolean)

    /** 连接失败，异常[cause] */
    fun onConnectFailure(cause: Throwable)

    /** 连接丢失（连上后断开），异常[cause] */
    fun onConnectionLost(cause: Throwable)
}